plugins/modules/storage_box.py

```diff
@@ -0,0 +1,609 @@
+
+# Copyright: (c) 2025, Hetzner Cloud GmbH <info@hetzner-cloud.de>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+
+from __future__ import annotations
+
+DOCUMENTATION = """
+---
+module: storage_box
+
+short_description: Create and manage Storage Boxes in Hetzner.
+
+description:
+    - Create, update and delete Storage Boxes in Hetzner.
+    - See the L(Storage Boxes API documentation,https://docs.hetzner.cloud/reference/hetzner#storage-boxes) for more details.
+
+author:
+    - Jonas Lammler (@jooola)
+
+options:
+    id:
+        description:
+            - ID of the Storage Box to manage.
+            - Required if no Storage Box O(name) is given.
+        type: int
+    name:
+        description:
+            - Name of the Storage Box to manage.
+            - Required if no Storage Box O(id) is given.
+            - Required if the Storage Box does not exist.
+        type: str
+    storage_box_type:
+        description:
+            - Name or ID of the Storage Box Type for the Storage Box.
+            - Required if the Storage Box does not exist.
+        type: str
+    location:
+        description:
+            - Name or ID of the Location for the Storage Box.
+            - Required if the Storage Box does not exist.
+        type: str
+    password:
+        description:
+            - Password for the Storage Box.
+            - Required if the Storage Box does not exist.
+        type: str
+    labels:
+        description:
+            - User-defined labels (key-value pairs) for the Storage Box.
+        type: dict
+    ssh_keys:
+        description:
+            - SSH public keys in OpenSSH format to inject into the Storage Box.
+        type: list
+        elements: str
+    access_settings:
+        description:
+            - Access settings of the Storage Box.
+        type: dict
+        suboptions:
+            reachable_externally:
+                description:
+                    - Whether access from outside the Hetzner network is allowed.
+                type: bool
+                default: false
+            samba_enabled:
+                description:
+                    - Whether the Samba subsystem is enabled.
+                type: bool
+                default: false
+            ssh_enabled:
+                description:
+                    - Whether the SSH subsystem is enabled.
+                type: bool
+                default: false
+            webdav_enabled:
+                description:
+                    - Whether the WebDAV subsystem is enabled.
+                type: bool
+                default: false
+            zfs_enabled:
+                description:
+                    - Whether the ZFS snapshot folder is visible.
+                type: bool
+                default: false
+    snapshot_plan:
+        description:
+            - Snapshot plan of the Storage Box.
+            - Use null to disabled the snapshot plan.
```

Comment:
```suggestion
            - Use null to disable the snapshot plan.
```

Can this differentiate between `null` (remove existing) and `undefined` (do nothing)? I assume yes, otherwise the examples below do not make sense.